### PR TITLE
Fixing #2592

### PR DESCRIPTION
**Pull request by Bricio**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fixing #2592 

Removed beeg.porn from valid url ( this domain doesn't host any video, it only links to others sites)
Removed tests URLs that are using links that returns 404 on browser.

Almost all code was not working, so i rewrote it.
I didn't use extract_m3u8_formats because i can't figured out how to process mp4 extension as m3u8 and generate mp4 file.
If there is a way, or better way, please enlighten me.

### CODE REVIEW

File: yt_dlp/extractor/beeg.py [hunk 1]
```diff
@@ -36,78 +50,54 @@ class BeegIE(InfoExtractor):
...
+            video_id, 'Downloading JSON for %s' % video_id)
 
-        for api_path in ('', 'api.'):
-            video = self._download_json(
-                'https://%sbeeg.com/api/v6/%s/video/%s'
-                % (api_path, beeg_version, video_id), video_id,
-                fatal=api_path == 'api.', query=query)
-            if video:
-                break
+        resources = video['file']['hls_resources'] if traverse_obj(video, ('file', 'hls_resources')) else video['fc_facts'][0]['hls_resources']
```
Comment (pukkandan) [inline yt_dlp/extractor/beeg.py:64]:
```suggestion
        resources = traverse_obj(video, ('file', 'hls_resources'), ('fc_facts', 0, 'hls_resources'))
```

File: yt_dlp/extractor/beeg.py [hunk 2]
```diff
@@ -36,78 +50,54 @@ class BeegIE(InfoExtractor):
...
 
         return {
             'id': video_id,
             'display_id': display_id,
             'title': title,
             'description': description,
-            'series': series,
             'timestamp': timestamp,
             'duration': duration,
             'tags': tags,
```
Comment (pukkandan) [inline yt_dlp/extractor/beeg.py:103]:
```suggestion
        return {
            'id': video_id,
            'display_id': traverse_obj(video, ('fc_facts', 0, 'id')),
            'title': traverse_obj(video, ('file', 'stuff', 'sf_name')),
            'description': traverse_obj(video, ('file', 'stuff', 'sf_story')),
            'timestamp': unified_timestamp(traverse_obj(video, ('fc_facts', 0, 'fc_created'))),
            'duration': int_or_none(traverse_obj(video, ('file', 'fl_duration'))),
            'tags': traverse_obj(video, ('tags', ..., 'tg_name')),
```

File: yt_dlp/extractor/beeg.py [hunk 3]
```diff
@@ -36,78 +50,54 @@ class BeegIE(InfoExtractor):
...
+        for format_id, video_uri in resources.items():
+            if not video_uri:
                 continue
             height = self._search_regex(
-                r'^(\d+)[pP]$', format_id, 'height', default=None)
+                r'fl_cdn_(\d+)', format_id, 'height', default=None)
             if not height:
                 continue
+#                formats.extend(self._extract_m3u8_formats(
+#                'https://video.beeg.com/' + video_uri, video_id, ext='m3u8', m3u8_id='hls'))
```
Comment (pukkandan) [inline yt_dlp/extractor/beeg.py:75]: What happens when you use this code?

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/beeg.py [hunk 4]
```diff
@@ -6,27 +6,41 @@
 )
 from ..utils import (
     int_or_none,
-    parse_qs,
     unified_timestamp,
+    traverse_obj,
```
Comment (pukkandan) [inline yt_dlp/extractor/beeg.py:10]: sort

File: yt_dlp/extractor/beeg.py (hunk 3 above)
Comment (Bricio) [reply yt_dlp/extractor/beeg.py:75]: wow, it works, i don't know what i did wrong yesterday.

but, it doesn't return the best video + audio.
listing formats we can see the difference.
with extract_m3u8_formats
image
generated video.
image

Manual extract.
image
generated video
image

Comment (pukkandan) [reply yt_dlp/extractor/beeg.py]: You can manually fill in the `height`/`width` in the formats that extract_m3u8 returns

File: yt_dlp/extractor/beeg.py [hunk 5]
```diff
@@ -84,23 +76,12 @@ def _real_extract(self, url):
...
-        duration = int_or_none(traverse_obj(video, ('file', 'fl_duration')))
-
-        tags = [tag.get('tg_name') for tag in video['tags']] if video.get('tags') else None
-
         return {
             'id': video_id,
-            'display_id': display_id,
-            'title': title,
-            'description': description,
-            'timestamp': timestamp,
```
Comment (pukkandan) [inline yt_dlp/extractor/beeg.py:111]: Why did you remove timestamp and display_id!?

Comment (Bricio) [reply yt_dlp/extractor/beeg.py:111]: Because It gets that from the first item of Json. ([fc_facts][0][id], or fc_created) it works when they are immutable. 

image

But for some videos this items changes. And we don't get a reliable information.

For example:
Yesterday the first item of the same video (https://beeg.com/-0599050563103750) had the id 2569975.
Today the first item is 2569969, and the 2569975 is on second item.
image

I removed it from return because tests keeps failing for the same correct video.

If there is a way to ignore these keys from tests and keep it in return, it is ok to bring it back.

Comment (pukkandan) [reply yt_dlp/extractor/beeg.py:111]: No, dont return wrong values! You can check the id of each matching entry and return if it matches. But I'll let you decide whether to implement it

Comment (Bricio) [reply yt_dlp/extractor/beeg.py:111]: I sorted items to get always the lower id, probably, the first (oldest) item. 
Now it should return always the same display_id and timestamp.

File: yt_dlp/extractor/beeg.py [hunk 6]
```diff
@@ -36,81 +48,49 @@ class BeegIE(InfoExtractor):
...
                 'height': int(height),
             })
-        self._sort_formats(formats)
-
-        title = video['title']
-        video_id = compat_str(video.get('id') or video_id)
-        display_id = video.get('code')
-        description = video.get('desc')
-        series = video.get('ps_name')
+            formats.extend(format)
```
Comment (pukkandan) [inline yt_dlp/extractor/beeg.py:82]:
```suggestion
            height = int_or_none(self._search_regex(r'fl_cdn_(\d+)', format_id, 'height', default=None))
            current_formats = self._extract_m3u8_formats(f'https://video.beeg.com/{video_uri}', video_id, ext='mp4', m3u8_id=str(height))
            for f in current_formats:
                f['height'] = height
            formats.extend(current_formats)

File: yt_dlp/extractor/beeg.py [hunk 7]
```diff
@@ -36,81 +48,49 @@ class BeegIE(InfoExtractor):
...
-                    's': t[0],
-                    'e': t[1],
-                })
-        else:
-            query = {'v': 1}
+        fc_facts = video.get('fc_facts')
+        firstKey = None
+        for i in range(len(fc_facts)):
+            if not firstKey or fc_facts[i - 1]['id'] < fc_facts[firstKey]['id']:
+                firstKey = i - 1
```
Comment (pukkandan) [inline yt_dlp/extractor/beeg.py:66]: I didnt test,  but pretty sure this is wrong

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/beeg.py (hunk 7 above)
Comment (Bricio) [reply yt_dlp/extractor/beeg.py:66]: Yea, there was 2 problems, 
i forgot it starts with 0, and i forgot list keys is not the same for multiple iterations on same code. changed the code to save first fact not first key.
